djstripe/settings.py

```diff
@@ -9,224 +9,276 @@
 
 from .checks import validate_stripe_api_version
 
-DEFAULT_STRIPE_API_VERSION = "2020-08-27"
-
-
-def get_callback_function(setting_name, default=None):
-    """
-    Resolve a callback function based on a setting name.
-
-    If the setting value isn't set, default is returned.  If the setting value
-    is already a callable function, that value is used - If the setting value
-    is a string, an attempt is made to import it.  Anything else will result in
-    a failed import causing ImportError to be raised.
-
-    :param setting_name: The name of the setting to resolve a callback from.
-    :type setting_name: string (``str``/``unicode``)
-    :param default: The default to return if setting isn't populated.
-    :type default: ``bool``
-    :returns: The resolved callback function (if any).
-    :type: ``callable``
-    """
-    func = getattr(settings, setting_name, None)
-    if not func:
-        return default
-
-    if callable(func):
-        return func
-
-    if isinstance(func, str):
-        func = import_string(func)
-
-    if not callable(func):
-        raise ImproperlyConfigured("{name} must be callable.".format(name=setting_name))
-
-    return func
-
-
-subscriber_request_callback = get_callback_function(
-    "DJSTRIPE_SUBSCRIBER_MODEL_REQUEST_CALLBACK", default=(lambda request: request.user)
-)
-
-
-def _get_idempotency_key(object_type, action, livemode):
-    from .models import IdempotencyKey
 
-    action = "{}:{}".format(object_type, action)
-    idempotency_key, _created = IdempotencyKey.objects.get_or_create(
-        action=action, livemode=livemode
+class DjstripeSettings:
+
+    DEFAULT_STRIPE_API_VERSION = "2020-08-27"
+
+    ZERO_DECIMAL_CURRENCIES = set(
+        [
+            "bif",
+            "clp",
+            "djf",
+            "gnf",
+            "jpy",
+            "kmf",
+            "krw",
+            "mga",
+            "pyg",
+            "rwf",
+            "vnd",
+            "vuv",
+            "xaf",
+            "xof",
+            "xpf",
+        ]
     )
-    return str(idempotency_key.uuid)
 
+    def __init__(self):
+        # Set STRIPE_API_HOST if you want to use a different Stripe API server
+        # Example: https://github.com/stripe/stripe-mock
+        if hasattr(settings, "STRIPE_API_HOST"):
+            stripe.api_base = getattr(settings, "STRIPE_API_HOST")
 
-get_idempotency_key = get_callback_function(
-    "DJSTRIPE_IDEMPOTENCY_KEY_CALLBACK", _get_idempotency_key
-)
+    # generic setter and deleter methods to ensure object patching works
+    def __setattr__(self, name, value):
+        self.__dict__["name"] = value
```

Comment:
@arnav13081994 is this meant to be ``"name"`` instead of ``name``? Same for the ``__delattr__`` method below.

Reply:
You are right @kavdev I have raised a [PR](https://github.com/dj-stripe/dj-stripe/pull/1399) fixing that and also added a test to make sure this never happens again.